### PR TITLE
Add function comments where missing

### PR DESCRIPTION
**Pull request by sirkrypt0**
This adds a lot of function comments whereever they were missing,
mostly on static functions.

### CODE REVIEW

**Review by frcroth — APPROVED** [review]

File: include/internal.h [hunk 1]
```diff
@@ -13,7 +13,28 @@
...
+
+/**
+ * @brief Ensure that the device is correctly set up to transmit data and
+ *        call the device's transport transmit function.
+ * 
+ * @param d A pointer to the FIDO device.
+ * @param cmd The CTAP command to transmit.
+ * @param buf A pointer to the source buffer.
+ * @param len The size of the source buffer.
+ * @return int FIDO_OK if the operation was successful.
```
Comment (felix-gohla) [inline include/internal.h:36]:
```suggestion
 * @return int FIDO_OK if the write operation was successful.
```

File: include/internal.h [hunk 2]
```diff
@@ -13,7 +13,28 @@
...
 
+/**
+ * @brief Ensure that the device is correctly set up to receive data and
+ *        call the device's transport receive function.
+ * 
+ * @param d A pointer to the FIDO device.
+ * @param cmd The CTAP command to receive data from.
+ * @param buf A pointer to the destination buffer.
+ * @param len The size of the destination buffer.
+ * @return int FIDO_OK if the operation was successful.
```
Comment (felix-gohla) [inline include/internal.h:24]:
```suggestion
 * @return int FIDO_OK if the read operation was successful.
```

File: src/assertion.c [hunk 3]
```diff
@@ -112,6 +112,14 @@ static const uint8_t KEY_TYPE[] PROGMEM_MARKER = "type";
 static const uint8_t KEY_TYPE_PUBLIC_KEY[] PROGMEM_MARKER = "public-key";
 static const uint8_t KEY_ID[] PROGMEM_MARKER = "id";
 
+/**
+ * @brief CBOR decode the credential's data such as its type and id.
```
Comment (felix-gohla) [inline src/assertion.c:116]:
```suggestion
 * @brief CBOR decode the credential's data such as its type and ID.
```

Comment (felix-gohla) [reply src/assertion.c:116]: Maybe also add a link to the standard.

File: src/assertion.c [hunk 4]
```diff
@@ -287,6 +326,14 @@ static int fido_dev_get_assert_tx(
     return ret;
 }
 
+/**
+ * @brief Receive the response data from the authenticator and parse the CBOR into the reply.
```
Comment (felix-gohla) [inline src/assertion.c:330]:
```suggestion
 * @brief Receive the response data from the authenticator and parse the authenticator's response into the reply.
```

**Comment by felix-gohla** [comment]
Could you please add a license header to `crypto.h`.

**Review by felix-gohla — CHANGES_REQUESTED** [review]

**Review by felix-gohla — APPROVED** [review]

**Pull request merged**